lib/mongo.ex

```diff
@@ -790,6 +795,92 @@ defmodule Mongo do
     bangify(update_many(topology_pid, coll, filter, update, opts))
   end
 
+  @doc """
+  Returns a cursor to enumerate all indexes
+
+  :full returns the full index information instead of the name only
+  """
+  @spec list_indexes(GenServer.server, String.t, Keyword.t) :: cursor
+  def list_indexes(topology_pid, coll, opts \\ []) do
+
+    with {:ok, conn, _, _} <- Mongo.select_server(topology_pid, :read) do
+
+      c = aggregation_cursor(conn, "$cmd", [listIndexes: coll], nil, [])
+
+      case Keyword.get(opts, :full, false) do
+        true -> c
+        _    -> Stream.map(c, fn %{"name" => name } -> name end)
+      end
+
+    end
+  end
+
+  @doc """
+  Getting Collection Names
+  """
+  @spec show_collections(GenServer.server, Keyword.t) :: cursor
+  def show_collections(topology_pid, opts \\ []) do
+
+    #
+    # todo:
+    #    listCollections can be run on a secondary
+    #    querying system.namespaces on a secondary requires slaveOkay to be set.
+    #    Drivers MUST run listCollections on the primary node in "replicaset" mode, unless directly connected to a
+    #    secondary node in "standalone" mode.
+    #
+    #
+    wv_query = %Query{action: :wire_version}
+
+    with {:ok, conn, _, _} <- select_server(topology_pid, :read, opts),
+         {:ok, version}    <- DBConnection.execute(conn, wv_query, [], defaults(opts)) do
+
+      case version do
+        v when v > 2 -> show_collections_v3(conn, opts)
+        _            -> show_collections_v2(topology_pid)
```

Comment:
I don't think we need to worry about wire version <= 2. Version 3 was introduced with mongo version 2.7.7, which we don't support. We only support 3.0+ right now.

Reply:
Ok, I will remove support for < 3.0